src/esp/assets/ResourceManager.cpp

```diff
@@ -1591,8 +1616,76 @@ int ResourceManager::loadNavMeshVisualization(esp::nav::PathFinder& pathFinder,
   return navMeshPrimitiveID;
 }  // ResourceManager::loadNavMeshVisualization
 
+namespace {
+
+/**
+ * @brief Make sure we do not add an attribute that already exists in source
+ * material - this will assert.
+ * @tparam the type of the value to be added
+ * @param material Source material for new attributes array
+ * @param newAttributes The array of attributes to be used to build a new
+ * Mn::Trade::MaterialData.
+ * @param matAttr The Mn::Trade::MaterialAttribute tag describing the attribute
+ * to add
+ * @param value A reference to the value to add.
+ */
+template <typename T>
+void safeAppend(
+    const Cr::Containers::Optional<Mn::Trade::MaterialData>& material,
```

Comment:
Again the optional here. Please don't :)

Same with `compareShaderTypeToMnMatType()`.

Reply:
We get an optional from the importer, are you saying deref it for these calls?

Reply:
Yes, the intended design / workflow is basically like this:

- the importer returns an Optional in order to signal if importing failed for some reason (for example when the glTF material was referencing a texture ID that's out of bounds)
- to avoid crazy complex error handling, not all importer APIs return an Optional, for example the various `fooCount()` and `fooName()` always return a valid value -- but when an API returns an Optional you're always expected to perform error handling
- when interacting with the importer, the best way is to immediately check the returned Optional and handle the error (for example -- replacing a material with a placeholder one, for a mesh just not rendering anything, for a missing image render as if it would be just colored...)
- and then, if not null, you dereference the Optional and then pass around just the underlying type